gallery_dl/extractor/poipiku.py

```diff
@@ -59,7 +60,7 @@ def items(self):
                     "//img.", "//img-org.", 1)
                 yield Message.Url, url, text.nameext_from_url(url, post)
 
-            if not extr('> show all', '<'):
+            if not re.search(r">\s*show all[\s\S]+?<", page):
```

Comment:
```suggestion
            if not extr(' show all(+', '<'):
```

No need for regex. Updating the searched string to this seems to work for both old and new "show all" buttons.

And regarding regular expressions in general: You should pre-compile them and, if possible, use a reference to the method that gets used.

```
    search_show_all = re.compile(r">\s*show all[\s\S]+?<").search
...
            if not search_show_all(page):
```

> `[\s\S]`

Isn't this the same as `.`, i.e. it matches everything?

Reply:
> Isn't this the same as ., i.e. it matches everything?

No,
```python
>>> print(re.search(r'te.*st', 'te \n st'))
None
>>> print(re.search(r'te[\s\S]*st', 'te \n st'))
<re.Match object; span=(0, 7), match='te \n st'>
```